### ISSUE TITLE
retrieve lane change state when lane change mode is 0

### ISSUE THREAD

**Issue by michele-segata** [DLR-TS/sumo#3810]
Hi all,
I've noticed than when setting lane change mode to 0 (to keep a vehicle in its current lane) the getLaneChangeState() API only returns whether the vehicle is blocked, but not whether the vehicle should change lane (speed gain / keep right) as the cancelRequest method changes the state to avoid the lane change (as requested by the lane change mode 0).
Is that possible to retrieve the decision computed by the lane change model even if this won't be applied?
Thanks!

**Comment by namdre** [comment]
I see two ways to fix this: 
1) remember all the cancelled requests (somewhat tricky to implement)
2) implement a new API that can retrieve the state for all possible motivations independent of each other (#3812) by calling the lane change model directly via TraCI (also quite a bit of implementation effort but potentially more useful)

Unfortunately, I don't think I'll have the time for either solution very soon.

**Comment by michele-segata** [comment]
I was looking into option 1. I made some quick tests that seem to work, but maybe you'll immediately spot something wrong in this idea. What about having an additional state variable which stores the state independently from what `cancelRequest()` does? I wrote a very simple patch:

[lc-state.txt](https://github.com/DLR-TS/sumo/files/1737319/lc-state.txt)

The patch still misses a TraCI query method (and probably changes to other LC models), but it is just to show you the idea.
